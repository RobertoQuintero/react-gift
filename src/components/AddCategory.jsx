import PropTypes from 'prop-types'
import { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange=(e)=>{
    setInputValue(e.target.value)

  }

  const onSubmit=(e)=>{
    console.log('Hola desde onSubmit')
    e.preventDefault();
    if(inputValue.trim().length<=1)return
    onNewCategory(inputValue.trim())
    setInputValue('')
  }
  return (
    <form onSubmit={onSubmit} aria-label='form'>

      <input 
      type="text" 
      placeholder='Buscar gifs'
      value={inputValue}
      onChange={onInputChange}
      />
    </form>
  )
}


AddCategory.propTypes={
  onNewCategory:PropTypes.func.isRequired
}