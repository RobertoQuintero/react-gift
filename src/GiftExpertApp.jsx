import { useState } from "react"
import { AddCategory,GifGrid } from "./components"

export const GiftExpertApp = () => {

  const [categories, setCategories] = useState([])

  const onAddCategory=(newCategory)=>{
    if(categories.includes(newCategory))return
    // setCategories(categories.concat('valorant'))
    // setCategories(cat=>cat.concat('valorant'))
    setCategories(cat=>[newCategory,...cat])
  }

  return (
    <>
    <h1>GifExpertApp</h1>
    <AddCategory onNewCategory={onAddCategory}/>
    <button onClick={onAddCategory} aria-label='btn-add'>Agregar</button>
      {categories.map(c=>
        <GifGrid key={c} category={c}/>
        )}
    </>
  )
}
