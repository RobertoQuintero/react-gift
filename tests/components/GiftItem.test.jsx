import { render,screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('prueba en GifItem', () => {
  const url='https://onepunch.com/images/12345abdc'
  const title= 'One Punch'

  test('evaluar snapshot', () => { 
    const {container}=render(<GifItem url={url} title={title}/>)
    expect(container).toMatchSnapshot()
  })

  test('debe mostar la imagen con el URL y el ALT indicado', () => { 

    render(<GifItem url={url} title={title} />)
    // screen.debug()
    // expect(screen.getByRole('img').src).toBe(url)
    // expect(screen.getByRole('img').alt).toBe(title)
    
    const {src,alt}= screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(alt)
  })
  
  test('debe mostar el titulo', () => { 
    
    render(<GifItem url={url} title={title} />)
    //despues de l render utilizamos screen para referirnos al componente
    expect(screen.getByText(title)).toBeTruthy()


  
  })

  

  
  
  
})


