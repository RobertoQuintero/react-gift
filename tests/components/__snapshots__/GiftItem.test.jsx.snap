// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en GifItem evaluar snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://onepunch.com/images/12345abdc"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
